File: src/main/java/de/axelrindle/broadcasterplugin/Formatter.java
package de.axelrindle.broadcasterplugin;

import org.apache.commons.lang.StringUtils;
import org.bukkit.ChatColor;
import org.bukkit.plugin.java.JavaPlugin;

import java.util.List;

/**
 * Helper-class for formatting messages.
 */
class Formatter {

    /**
     * Formats a message according to used placeholders and color codes.
     *
     * @param plugin The {@link JavaPlugin} to get config values from.
     * @param message The message to format.
     *
     * @return A formatted message.
     * @see ChatColor#translateAlternateColorCodes(char, String)
     * @see JavaPlugin#getConfig()
     */
    @SuppressWarnings("unchecked")
    static String format(JavaPlugin plugin, String message) {
        // format placeholders
        List<List<String>> placeholders = (List<List<String>>) plugin.getConfig().getList("Placeholders");
        for (List<String> holder : placeholders) {
            String key = holder.get(0);
            String value = holder.get(1);

            message = StringUtils.replace(message, key, value);
        }

        // translate color codes
        message = ChatColor.translateAlternateColorCodes('&', message);

        return message;
    }

    static String formatColors(String message) {
        return ChatColor.translateAlternateColorCodes('&', message);
    }
}

File: src/main/kotlin/de/axelrindle/broadcaster/Formatter.kt
package de.axelrindle.broadcaster

import de.axelrindle.pocketknife.util.ChatUtils.formatColors
import org.apache.commons.lang.StringUtils
import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.plugin.java.JavaPlugin

/**
 * Helper-class for formatting messages.
 */
object Formatter {

    /**
     * Formats a message according to used placeholders and color codes.
     *
     * @param plugin The [Broadcaster] instance to get config values from.
     * @param msg The message to format.
     *
     * @return A formatted message.
     * @see ChatColor.translateAlternateColorCodes
     * @see JavaPlugin.getConfig
     */
    @Suppress("UNCHECKED_CAST")
    fun format(plugin: Broadcaster, msg: String): String {
        var message = msg
        // format placeholders
        val placeholders = plugin.config.access("config")!!
                .getList("Placeholders") as List<List<String>>

        for (holder in placeholders) {
            val key = holder[0]
            val value = holder[1]

            message = StringUtils.replace(message, key, value)
        }
        message = formatPredefinedPlaceholders(message)

        // translate color codes
        message = formatColors(message)

        return message
    }

    private fun formatPredefinedPlaceholders(msg: String): String {
        var message = msg

        // line break
        message = StringUtils.replace(message, "%n", "\n")

        // online players
        message = StringUtils.replace(message, "%online_players%", Bukkit.getServer().onlinePlayers.size.toString())

        // max players
        message = StringUtils.replace(message, "%max_players%", Bukkit.getServer().maxPlayers.toString())

        return message
    }
}